tests/std/tests/P0896R4_ranges_alg_search_n/test.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::search_n(borrowed<false>{}, 13, 42)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::search_n(borrowed<true>{}, 42, 13)), ranges::subrange<int*>>);
+
+using P = pair<int, int>;
+
+struct instantiator {
+    static constexpr array<P, 11> pairs = {
+        P{0, 42}, P{1, 42}, P{1, 42}, P{0, 42}, P{1, 42}, P{1, 42}, P{0, 42}, P{1, 42}, P{1, 42}, P{1, 42}, P{0, 42}};

```

Comment:
I am somewhat concerned that there's significant control flow in the algorithm, involving skipping elements, but these hardcoded tests don't seem to exercise every possible branch in that logic. `search_n` has historically been a tricky algorithm to get right - there was a bug in the classic algorithm fixed just over a decade ago:

https://github.com/microsoft/STL/blob/4deaf6cb27716170e251ed89fd0d710235ea6995/tests/std/tests/Dev08_584299_search_n/test.cpp#L8-L14

Should we have an exhaustive or randomized test? (I would be fine with filing a `test` issue, I think.)

Reply:
The tests cover both sides of each branch and exercise each line of the implementation at least once. That said, I think a randomized test would be nice - filed #933.